### ISSUE TITLE
Unable to connect to redis outside the k8s cluster

### ISSUE THREAD

**Issue by lys0224** [istio/istio#24544]
**Bug description**

Everything worked well on the environment with istio 1.4.2, but when I update Istio from 1.4.2 to 1.5.4 ,there is a problem.
Some microservices that need a connection to Redis throw Exceptions(Redis is outside the k8s cluster).
I tried to configure a service entry，it doesn't work.
I tried to update the istio to 1.6.1,it doesn't work too.
But when the sidecar  injection is cancelled, the microservices work normally!

The exception thrown by the microservice is :
```
2020-06-09 15:01:26.444 DEBUG 1 --- [ioEventLoop-4-1] io.lettuce.core.protocol.CommandHandler  : [channel=0x439cb4da, /172.20.179.156:47854 -> /192.168.13.56:6379, chid=0x1] Unexpected exception during request: java.io.IOException: Connection reset by peer

java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at io.netty.buffer.PooledUnsafeDirectByteBuf.setBytes(PooledUnsafeDirectByteBuf.java:288)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1106)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:343)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:123)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:645)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:580)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:497)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:459)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:138)
	at java.lang.Thread.run(Thread.java:748)

2020-06-09 15:01:26.445 ERROR 1 --- [nio-9000-exec-7] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception

org.springframework.data.redis.RedisConnectionFailureException: Unable to connect to Redis; nested exception is io.lettuce.core.RedisConnectionException: Unable to connect to 192.168.13.56:6379
	at org.springframework.data.redis.connection.lettuce.LettuceConnectionFactory$SharedConnection.getNativeConnection(LettuceConnectionFactory.java:1106)
	at org.springframework.data.redis.connection.lettuce.LettuceConnectionFactory$SharedConnection.getConnection(LettuceConnectionFactory.java:1085)
	at org.springframework.data.redis.connection.lettuce.LettuceConnectionFactory.getSharedConnection(LettuceConnectionFactory.java:866)
......
```

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
The pod injected into the sidecar should be able to connect to redis outside the k8s cluster normally.

**Steps to reproduce the bug**
A java application use Spring RedisTemplate to connect the redis.
A standalone Redis deploied outside the k8s cluster.
A Service Entry 
```
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: test-redis
  namespace: test
spec:
  endpoints:
  - address: 192.168.13.56
  hosts:
  - test-redis.local
  location: MESH_EXTERNAL
  ports:
  - name: redis6379
    number: 6379
    protocol: TCP
  resolution: STATIC
```

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
1.5.4/1.6.1

**How was Istio installed?**
istioctl, defalut profile

**Environment where bug was observed (cloud vendor, OS, etc)**
Linux Cluster,centos7，k8s 1.16

**Comment by esnible** [comment]
Could this be a duplicate of #12079 #14138 #14743 #15429 or #16078 ?  The Redis-specific issued are all closed but they may suggest a solution for this problem.

**Comment by lys0224** [comment]
> Could this be a duplicate of #12079 #14138 #14743 #15429 or #16078 ? The Redis-specific issued are all closed but they may suggest a solution for this problem.

None of this worked

**Comment by lys0224** [comment]
I uninstalled version 1.5/1.6 of istio，and I reinstalled version 1.4.9 ,all microservices work fine.The redis connection is normal.

**Comment by lys0224** [comment]
I tried 1.6.3,it doesn't work too.

**Comment by lys0224** [comment]
> Could this be a duplicate of #12079 #14138 #14743 #15429 or #16078 ? The Redis-specific issued are all closed but they may suggest a solution for this problem.

I uninstalled version 1.5/1.6 of istio，and I reinstalled version 1.4.9 ,all microservices work fine.The redis connection is normal.
I tried newest 1.6.3, it doesn't work too.
I tried intall statefulSets of redis in kubernetes ,the connection is normal with 1.6.3.
It seems that the connection to external redis fails when the istio is over version 1.5.
